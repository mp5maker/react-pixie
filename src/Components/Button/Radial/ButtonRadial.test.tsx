import React from 'react'
import { shallow, render, mount } from 'enzyme'
import sinon from 'sinon'

import { ButtonRadial } from '.'

it('it should shallow properly', () => {
    const wrapper = shallow(
        <ButtonRadial
            style={{}}
            onClick={() => {}}>
            Sample Button
        </ButtonRadial>
    )

    expect(wrapper).toMatchSnapshot()
})

it('it should render properly', () => {
    const wrapper = render(
        <ButtonRadial
            style={{}}
            onClick={() => {}}>
            Sample Button
        </ButtonRadial>
    )

    expect(wrapper).toMatchSnapshot()
})

it('it should mount properly', () => {
    const spy = sinon.spy()

    const wrapper = mount(
        <ButtonRadial
            style={{}}
            onClick={spy}>
            Sample Button
        </ButtonRadial>
    )

    wrapper
        .find('.button-radial-container')
        .first()
        .simulate('click')

    expect(spy.calledOnce).toBe(true)
})
