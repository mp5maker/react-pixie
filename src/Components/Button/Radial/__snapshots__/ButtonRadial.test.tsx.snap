// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render properly 1`] = `
<button
  class="button-radial-container"
>
  Sample Button
</button>
`;

exports[`it should shallow properly 1`] = `
<ForwardRef(MotionComponent)
  className="button-radial-container"
  onClick={[Function]}
  style={Object {}}
>
  Sample Button
</ForwardRef(MotionComponent)>
`;
